diff --git a/plugins/strtotime.inc b/plugins/strtotime.inc
index a4fb1e9..28134a8 100644
--- a/plugins/strtotime.inc
+++ b/plugins/strtotime.inc
@@ -22,5 +22,21 @@ function feeds_tamper_strtotime_form($importer, $element_key, $settings) {
 }
 
 function feeds_tamper_strtotime_callback($result, $item_key, $element_key, &$field, $settings, $source) {
-  $field = strtotime($field);
+  $field =  strtotime(feeds_tamper_danish_month_to_english($field));
 }
+function feeds_tamper_danish_month_to_english($str) {
+  $str = str_replace('januar', 'January', $str);
+  $str = str_replace('februar', 'February', $str);
+  $str = str_replace('marts', 'March', $str);
+  $str = str_replace('april', 'April', $str);
+  $str = str_replace('maj', 'May', $str);
+  $str = str_replace('juni', 'June', $str);
+  $str = str_replace('juli', 'July', $str);
+  $str = str_replace('august', 'August', $str);
+  $str = str_replace('september', 'September', $str);
+  $str = str_replace('oktober', 'October', $str);
+  $str = str_replace('november', 'November', $str);
+  $str = str_replace('december', 'December', $str);
+
+  return $str;
+}
\ No newline at end of file
