diff --git a/views_administration_content_referenced_by.features.inc b/views_administration_content_referenced_by.features.inc
index ff41be7..24a52d2 100644
--- a/views_administration_content_referenced_by.features.inc
+++ b/views_administration_content_referenced_by.features.inc
@@ -10,96 +10,3 @@
 function views_administration_content_referenced_by_views_api($module = NULL, $api = NULL) {
   return array("api" => "3.0");
 }
-
-/**
- * Implements hook_themekey_features_rule_chain().
- */
-function views_administration_content_referenced_by_themekey_features_rule_chain() {
-if (!defined('THEMEKEY_PAGECACHE_UNSUPPORTED')) {
-    define('THEMEKEY_PAGECACHE_UNSUPPORTED', 0);
-    define('THEMEKEY_PAGECACHE_SUPPORTED', 1);
-    define('THEMEKEY_PAGECACHE_TIMEBASED', 2);
-  }
-$rules = array(
-  0 => array(
-    'rule' => array(
-      'property' => 'drupal:path',
-      'operator' => '=',
-      'value' => 'node/%/faq-referenced-by',
-      'theme' => 'seven',
-      'enabled' => 1,
-      'wildcards' => array(),
-      'module' => 'views_administration_content_referenced_by',
-    ),
-    'string' => '"drupal:path = node/%/faq-referenced-by >>> seven"',
-    'childs' => array(),
-  ),
-  1 => array(
-    'rule' => array(
-      'property' => 'drupal:path',
-      'operator' => '=',
-      'value' => 'node/%/factbox-referenced-by',
-      'theme' => 'seven',
-      'enabled' => 1,
-      'wildcards' => array(),
-      'module' => 'views_administration_content_referenced_by',
-    ),
-    'string' => '"drupal:path = node/%/factbox-referenced-by >>> seven"',
-    'childs' => array(),
-  ),
-  2 => array(
-    'rule' => array(
-      'property' => 'drupal:path',
-      'operator' => '=',
-      'value' => 'node/%/selfservice-referenced-by',
-      'theme' => 'seven',
-      'enabled' => 1,
-      'wildcards' => array(),
-      'module' => 'views_administration_content_referenced_by',
-    ),
-    'string' => '"drupal:path = node/%/selfservice-referenced-by >>> seven"',
-    'childs' => array(),
-  ),
-  3 => array(
-    'rule' => array(
-      'property' => 'drupal:path',
-      'operator' => '=',
-      'value' => 'node/%/bannerlink-referenced-by',
-      'theme' => 'seven',
-      'enabled' => 1,
-      'wildcards' => array(),
-      'module' => 'views_administration_content_referenced_by',
-    ),
-    'string' => '"drupal:path = node/%/bannerlink-referenced-by >>> seven"',
-    'childs' => array(),
-  ),
-  4 => array(
-    'rule' => array(
-      'property' => 'drupal:path',
-      'operator' => '=',
-      'value' => 'node/%/phonebook-referenced-by',
-      'theme' => 'seven',
-      'enabled' => 1,
-      'wildcards' => array(),
-      'module' => 'views_administration_content_referenced_by',
-    ),
-    'string' => '"drupal:path = node/%/phonebook-referenced-by >>> seven"',
-    'childs' => array(),
-  ),
-  5 => array(
-    'rule' => array(
-      'property' => 'drupal:path',
-      'operator' => '=',
-      'value' => 'node/%/borgerdk-referenced-by',
-      'theme' => 'seven',
-      'enabled' => 1,
-      'wildcards' => array(),
-      'module' => 'views_administration_content_referenced_by',
-    ),
-    'string' => '"drupal:path = node/%/borgerdk-referenced-by >>> seven"',
-    'childs' => array(),
-  ),
-);
-
-return $rules;
-}
diff --git a/views_administration_content_referenced_by.info b/views_administration_content_referenced_by.info
index a38fc5c..bbdebff 100644
--- a/views_administration_content_referenced_by.info
+++ b/views_administration_content_referenced_by.info
@@ -4,16 +4,8 @@ core = 7.x
 package = Views
 version = 7.x-1.0
 project = views_administration_content_referenced_by
-dependencies[] = themekey
-dependencies[] = themekey_features
 dependencies[] = views
 features[ctools][] = views:views_default:3.0
 features[features_api][] = api:2
-features[themekey_features_rule_chain][] = drupal:path = node/%/bannerlink-referenced-by >>> seven
-features[themekey_features_rule_chain][] = drupal:path = node/%/borgerdk-referenced-by >>> seven
-features[themekey_features_rule_chain][] = drupal:path = node/%/factbox-referenced-by >>> seven
-features[themekey_features_rule_chain][] = drupal:path = node/%/faq-referenced-by >>> seven
-features[themekey_features_rule_chain][] = drupal:path = node/%/phonebook-referenced-by >>> seven
-features[themekey_features_rule_chain][] = drupal:path = node/%/selfservice-referenced-by >>> seven
 features[views_view][] = administration_content_referenced_by
 features_exclude[dependencies][ctools] = ctools
diff --git a/views_administration_content_referenced_by.module b/views_administration_content_referenced_by.module
index 48d5b5a..c432320 100644
--- a/views_administration_content_referenced_by.module
+++ b/views_administration_content_referenced_by.module
@@ -5,3 +5,17 @@
  */
 
 include_once 'views_administration_content_referenced_by.features.inc';
+
+/**
+ * Implements hook_admin_paths().
+ */
+function views_administration_content_referenced_by_admin_paths() {
+  return array(
+    'node/*/faq-referenced-by' => TRUE,
+    'node/*/factbox-referenced-by' => TRUE,
+    'node/*/selfservice-referenced-by' => TRUE,
+    'node/*/bannerlink-referenced-by' => TRUE,
+    'node/*/phonebook-referenced-by' => TRUE,
+    'node/*/borgerdk-referenced-by' => TRUE,
+  );
+}
