diff --git a/drupal/sites/default/default.settings.php b/drupal/sites/default/default.settings.php
index b044be3..f178ce6 100644
--- a/drupal/sites/default/default.settings.php
+++ b/drupal/sites/default/default.settings.php
@@ -628,3 +628,5 @@ $conf['404_fast_html'] = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML+RDFa 1.0//EN"
  * @see drupal_clean_css_identifier()
  */
 # $conf['allow_css_double_underscores'] = TRUE;
+
+$base_url = "http" . (isset($_SERVER["HTTPS"]) && strtolower($_SERVER["HTTPS"]) == "on" ? "s" : "") . "://" . $_SERVER["HTTP_HOST"];
