diff --git a/sites/default/default.settings.php b/sites/default/default.settings.php
index 7e36a4a..11a50e1 100644
--- a/sites/default/default.settings.php
+++ b/sites/default/default.settings.php
@@ -584,3 +584,5 @@ $conf['404_fast_html'] = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML+RDFa 1.0//EN"
  * Remove the leading hash sign to enable.
  */
 # $conf['theme_debug'] = TRUE;
+
+$base_url = "http" . (isset($_SERVER["HTTPS"]) && strtolower($_SERVER["HTTPS"]) == "on" ? "s" : "") . "://" . $_SERVER["HTTP_HOST"];
