diff --git a/mappers/text.inc b/mappers/text.inc
index 48447d7..a59a2c4 100644
--- a/mappers/text.inc
+++ b/mappers/text.inc
@@ -34,7 +34,7 @@ function text_feeds_processor_targets_alter(&$targets, $entity_type, $bundle_nam
  * Callback for mapping text fields.
  */
 function text_feeds_set_target($source, $entity, $target, $value) {
-  if (empty($value)) {
+  if ($value === NULL) {
     return;
   }
 
