From 2905a8ae5d32d5adf5e2b52c8fe5b01f6884f24f Mon Sep 17 00:00:00 2001
From: jbolduc <jbolduc@genuineinteractive.com>
Date: Wed, 15 Oct 2014 14:36:49 -0400
Subject: [PATCH] JJ-1997: 3rd party search plugin fix

---
 apachesolr_multilingual.module | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/apachesolr_multilingual.module b/apachesolr_multilingual.module
index 4208ac3..4ec74a8 100644
--- a/apachesolr_multilingual.module
+++ b/apachesolr_multilingual.module
@@ -506,7 +506,7 @@ function apachesolr_multilingual_apachesolr_query_alter($query) {
   $settings = $environment['conf']['apachesolr_multilingual_index_settings'];
   if ($settings['apachesolr_multilingual_index']) {
     $custom_settings = array();
-    if ('apachesolr_search_mlt' == $context['search_type']) {
+    if (isset($context['search_type']) && 'apachesolr_search_mlt' == $context['search_type']) {
       // Add multilingual settings to mlt blocks like we do for search pages.
       // @see https://drupal.org/node/2056055
       $mlt_block_settings = apachesolr_search_mlt_block_load($context['block_id']);
@@ -528,8 +528,15 @@ function apachesolr_multilingual_apachesolr_query_alter($query) {
     }
     // Add defaults for required settings to not yet saved multilingual custom settings.
     apachesolr_multilingual_search_page_settings_add_defaults($custom_settings);
-    $filter_languages = &drupal_static('apachesolr_multilingual_filter_languages_' . $context['page_id']);
+
+    if(!empty($context['page_id'])){
+      $filter_languages = &drupal_static('apachesolr_multilingual_filter_languages_' . $context['page_id']);
+    }else{
+      $filter_languages = &drupal_static('apachesolr_multilingual_filter_languages');
+    }
+
     $filter_languages = apachesolr_multilingual_get_language_filters_by_query($query);
+
     $languages = apachesolr_multilingual_language_list();
     if (empty($filter_languages) && $custom_settings['apachesolr_multilingual_auto_language_filter'] &&
       (!$custom_settings['apachesolr_multilingual_auto_language_filter_detachable'] ||
-- 
1.9.2

