diff --git a/views/media_browser_plus_views_handler_area_navigation.inc b/views/media_browser_plus_views_handler_area_navigation.inc
index 008794a..7ef7e8f 100644
--- a/views/media_browser_plus_views_handler_area_navigation.inc
+++ b/views/media_browser_plus_views_handler_area_navigation.inc
@@ -64,7 +64,16 @@ class media_browser_plus_views_handler_area_navigation extends views_handler_are
       $this->value = $root_folder->tid;
     }
     $this->ensure_my_table();
-    $this->query->add_where(1, "$this->table_alias.$this->real_field", $this->value, '=');
+    $folders = array($this->value);
+    if (isset($this->view->exposed_data['mbp_current_folder_include_subfolders']) && $this->view->exposed_data['mbp_current_folder_include_subfolders'] !== 0) {
+      $taxonomy_tree = taxonomy_get_tree(taxonomy_vocabulary_machine_name_load('media_folders')->vid, $this->value);
+      if (is_array($taxonomy_tree)) {
+        foreach ($taxonomy_tree as $term) {
+          $folders[] = $term->tid;
+        }
+      }
+    }
+    $this->query->add_where(1, "$this->table_alias.$this->real_field", $folders, 'IN');
   }
 
   /**
@@ -124,5 +133,12 @@ class media_browser_plus_views_handler_area_navigation extends views_handler_are
         'class' => array('mbp-selected-folder'),
       ),
     );
+
+    // Add a checkbox to include subfolders in the view results.
+    $form['mbp_current_folder_include_subfolders'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Include subfolders in results'),
+      '#default_value' => 0,
+    );
   }
 }
