diff --git a/admin_views_default/node.admin-content.inc b/admin_views_default/node.admin-content.inc
index b488a2c..a23f0e8 100644
--- a/admin_views_default/node.admin-content.inc
+++ b/admin_views_default/node.admin-content.inc
@@ -350,6 +350,24 @@ $handler->display->display_options['filters']['vid']['expose']['label'] = 'Vocab
 $handler->display->display_options['filters']['vid']['expose']['operator'] = 'vid_op';
 $handler->display->display_options['filters']['vid']['expose']['identifier'] = 'vid';
 $handler->display->display_options['filters']['vid']['expose']['remember'] = TRUE;
+/* Filter criterion: Domain Access: Domain ID */
+$handler->display->display_options['filters']['gid']['id'] = 'gid';
+$handler->display->display_options['filters']['gid']['table'] = 'domain_access';
+$handler->display->display_options['filters']['gid']['field'] = 'gid';
+$handler->display->display_options['filters']['gid']['exposed'] = TRUE;
+$handler->display->display_options['filters']['gid']['expose']['operator_id'] = 'gid_op';
+$handler->display->display_options['filters']['gid']['expose']['label'] = 'Domain';
+$handler->display->display_options['filters']['gid']['expose']['operator'] = 'gid_op';
+$handler->display->display_options['filters']['gid']['expose']['identifier'] = 'gid';
+$handler->display->display_options['filters']['gid']['expose']['remember'] = TRUE;
+$handler->display->display_options['filters']['gid']['expose']['remember_roles'] = array(
+  2 => '2',
+  1 => 0,
+  6 => 0,
+  3 => 0,
+  4 => 0,
+  5 => 0,
+);
 
 /* Display: System */
 $handler = $view->new_display('system', 'System', 'system_1');
@@ -383,5 +401,6 @@ $translatables['admin_views_node'] = array(
   t('Updated'),
   t('Operations'),
   t('Vocabulary'),
+  t('Domain'),
   t('System'),
 );
